File: Android source code/V3/DeviceListAdapter.java
package com.example.unity3dbluetoothplugin;

import android.bluetooth.BluetoothDevice;

import java.util.HashMap;
import java.util.Map;

public class DeviceListAdapter {
    public Map<String, BluetoothDevice> mDevicesMap = new HashMap<String, BluetoothDevice>();

    public boolean AddDevice(BluetoothDevice device) {
        if (this.mDevicesMap.get(device.getAddress()) != null) {
            return false;
        }

        this.mDevicesMap.put(device.getAddress(), device);
        return true;
    }

    public int getCount() {
        return this.mDevicesMap.size();
    }

    public BluetoothDevice getItem(String i) {
        return this.mDevicesMap.get(i);
    }

    public void clearAll(){
        this.mDevicesMap.clear();
    }
}

File: blueunity/src/main/java/com/benstudio/blueunity/DeviceListAdapter.kt
package com.benstudio.blueunity

import android.bluetooth.BluetoothDevice

class DeviceListAdapter {
    private val devicesMap: MutableMap<String, BluetoothDevice> = mutableMapOf()

    fun addDevice(device: BluetoothDevice): Boolean {
        if (devicesMap[device.address] != null) {
            return false
        }
        devicesMap[device.address] = device
        return true
    }

    fun getCount(): Int {
        return devicesMap.size
    }

    fun getItem(address: String): BluetoothDevice? {
        return devicesMap[address]
    }

    fun clearAll() {
        devicesMap.clear()
    }
}